ArgoProject/src/components/Camera.h

```diff
@@ -21,9 +21,11 @@ namespace app::comp
 	public: // Public Member Functions
 	public: // Public Static Variables
 	public: // Public Member Variables
-		math::Vector2f position;
+		math::Vector2f center;
 		math::Vector2f size;
 		std::optional<app::Entity> target;
+		std::optional<math::Rectf> clampRect;
+		std::optional<float> internalClampSize;
```

Comment:
```suggestion
		std::optional<math::Vector2f> internalClamp;
```
Needs to have different values for x-axis vs y-axis.